### DISCUSSION TITLE
How can I compile Akita and MGPUsim at the same time?? I can't get through errors with packages

### DISCUSSION THREAD

**Discussion by JunjoFor** [sarchlab/mgpusim#102] [Q&A] [answered]
Hi, I want to add in mem-trace the TLB access trace (with Vaddr). I found out that I need to modify the tlb.go file in akita to implement this. I can't compile MGPUsim with modified akita, I don't know how to do it. I tried to do a fork of akita and change all the implementations by my fork (changing also go.mod) but it throw an error about versions. I tried to clone my fork into a local folder, put replace statement in the go.mod but I still get errors. I tried modifying the go/pkg/github... to hardcode my github fork there, that didn't work either. 

Besides my plan to extend mem-trace I would like to research with MGPUsim so I will need to modify akita further.

I spend hours in this I don't know how to compile modifications in Akita with MGPUsim :( . I would love help in how to do it.

**Comment by syifan** [comment]
You can modify the `go.mod` file in the MGPUSim repo. There is a commented out line https://github.com/sarchlab/mgpusim/blob/v3/go.mod#L45. 

If you uncomment this line, you use your local version of Akita, which should be placed next to the MGPUSim repo.

**Reply by syifan** [reply, marked as answer]
I hope this added section in the readme file can address your question. https://github.com/sarchlab/mgpusim/pull/103/files

**Reply by syifan** [reply]
@JunjoFor Please let me know if the solution works for does not work for you.

**Reply by JunjoFor** [reply]
Yeah It work ! Thank you very much. I already implemented the L1-TLB traces
